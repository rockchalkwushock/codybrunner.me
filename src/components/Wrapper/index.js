import styled from 'styled-components'

const Wrapper = styled.div`
  border: 2px solid yellow;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 900px;
`

Wrapper.displayName = 'Wrapper'

export default Wrapper
