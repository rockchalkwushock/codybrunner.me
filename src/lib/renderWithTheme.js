import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
