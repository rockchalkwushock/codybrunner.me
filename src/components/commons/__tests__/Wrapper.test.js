import React from 'react'

import renderWithTheme from '../../../lib/renderWithTheme'
import Wrapper from '../Wrapper'

describe('Component: <Wrapper />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<Wrapper />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('background-color', '#7a9eb1')
    expect(wrapper).toHaveStyleRule('color', '#143a52')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('font-family', '"Raleway",sans-serif')
    expect(wrapper).toHaveStyleRule('font-size', '1rem')
    expect(wrapper).toHaveStyleRule('font-weight', '400')
    expect(wrapper).toHaveStyleRule('letter-spacing', '0px')
    expect(wrapper).toHaveStyleRule('line-height', '1.4')
    expect(wrapper).toHaveStyleRule('margin', '0 auto')
    expect(wrapper).toHaveStyleRule('min-height', '100vh')
    expect(wrapper).toHaveStyleRule('max-width', '900px')
  })
})
