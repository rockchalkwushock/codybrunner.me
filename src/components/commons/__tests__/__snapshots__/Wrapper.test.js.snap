// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Wrapper /> should render correctly 1`] = `
.c0 {
  background-color: #7a9eb1;
  color: #143a52;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: "Raleway",sans-serif;
  font-size: 1rem;
  font-weight: 400;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  line-height: 1.4;
  margin: 0 auto;
  min-height: 100vh;
  max-width: 900px;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "fg": "",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
      },
    }
  }
>
  <Wrapper>
    <div
      className="c0"
    />
  </Wrapper>
</ThemeProvider>
`;
